### DISCUSSION TITLE
facing an error

### DISCUSSION THREAD

**Discussion by msru** [IvanYashchuk/fenics-pymc3#6] [Q&A]
Hi Ivan,
I am a beginner to fenics-pymc3 and I have recently installed all the requirements. I tried to run your example but an error occurred as bellow. Would you please let me know haw to solve it?
Best,

~/anaconda3/envs/fenics2018/lib/python3.7/site-packages/fenics_adjoint/blocks/solving.py in _assemble_and_solve_adj_eq(self, dFdu_adj_form, dJdu, compute_bdy)
    117 
    118         if solver_method in lu_solver_methods:
--> 119             solver = self.backend.LUSolver(solver_method)
    120             solver_parameters = self.adj_kwargs.get("lu_solver", {})
    121         else:

TypeError: __init__(): incompatible constructor arguments. The following argument types are supported:
    1. dolfin.cpp.la.LUSolver()
    2. dolfin.cpp.la.LUSolver(A: dolfin.cpp.la.GenericLinearOperator, method: str='default')
    3. dolfin.cpp.la.LUSolver(comm: MPICommWrapper, A: dolfin.cpp.la.GenericLinearOperator, method: str='default')

Invoked with: 'default'

**Comment by IvanYashchuk** [comment]
Hello @msru!
Thanks for reporting the problem! I need to pin the requirements to make sure it all works smoothly. I will try to do that over the weekend.

**Reply by msru** [reply]
Hi, @IvanYashchuk 
Have you worked on that?

**Reply by IvanYashchuk** [reply]
Hey @msru, I'm sorry for the delay. What version of FEniCS/Dolfin are you using? I see that your environment name is `fenics2018`. Could you upgrade to the last release of FEniCS which is `2019.1.0`.
I just followed the following steps using [mamba ](https://github.com/mamba-org/mamba) alternative to conda.
```bash
mamba create -n fenicsproject -c conda-forge fenics superlu_dist=6.2.0=h050a5a1_4
conda activate fenicsproject
python -m pip install --upgrade pip
python -m pip install pytest
python -m pip install pytest-check
python -m pip install git+https://github.com/dolfin-adjoint/pyadjoint.git@master
python -m pip install git+https://github.com/IvanYashchuk/fecr.git@master
python -m pip install pymc3
python -m pip install git+https://github.com/IvanYashchuk/fenics-pymc3.git@master
```
Then all test pass when running
```bash
python -m pytest tests/fenics/
```
Note that the restriction of `superlu_dist` is required as recent builds of that package broke fenics from conda-forge.

**Reply by msru** [reply]
Thank you very much for your response. I will try it and let you know the result.

**Reply by msru** [reply]
Thanks. 
It worked.

**Comment by msru** [comment]
Dear @IvanYashchuk ,
I wanted to write a code to solve a Nonlinear-Poisson equation with combining Dirichlet and Neumann conditions. I am going to use it in your fenics-pymc3 approach. But at the first step, I have had problems in solving Nonlinear-Poisson equation with mixed conditions. Would you please kindly let me know what should I do with my code and how to resolve this error?
Attached you can find my code.
[Nonlinear-Poisson equation with Combining Dirichlet and Neumann conditions.md](https://github.com/IvanYashchuk/fenics-pymc3/files/8325395/Nonlinear-Poisson.equation.with.Combining.Dirichlet.and.Neumann.conditions.md)
